Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/EditorExtrinsicsCalibration.cs

```diff
@@ -0,0 +1,341 @@
+        private HeadsetCalibrationData headsetData = null;
+        private int nextArUcoImageId = 0;
+        private List<CalculatedCameraExtrinsics> cameraExtrinsics;
+        private CalculatedCameraExtrinsics globalExtrinsics;
+        private List<GameObject> parentVisuals = new List<GameObject>();
+
+#if UNITY_EDITOR
+        private void OnValidate()
+        {
+            FieldHelper.ValidateType<INetworkingService>(NetworkingService);
+            FieldHelper.ValidateType<IMatchMakingService>(MatchMakingService);
+        }
+
+        private void Start()
+        {
+            CalibrationDataHelper.Initialize(out var nextChessboardImageId, out nextArUcoImageId);
+            dslrIntrinsics = CalibrationDataHelper.LoadCameraIntrinsics(cameraIntrinsicsPath);
+            if (dslrIntrinsics == null)
+            {
+                throw new Exception("Failed to load the camera intrinsics file.");
+            }
+            else
+            {
+                Debug.Log($"Successfully loaded the provided camera intrinsics file: {dslrIntrinsics}");
+            }
+
+            calibration = new CalibrationAPI();
+
+            networkingService = NetworkingService as INetworkingService;
+            if (networkingService != null)
+            {
+                networkingService.DataReceived += OnDataReceived;
+            }
+
+            matchMakingService = MatchMakingService as IMatchMakingService;
+            if (matchMakingService != null)
+            {
+                matchMakingService.Connect();
+            }
+
+            if (headsetCalibration != null)
+            {
+                headsetCalibration.Updated += OnHeadsetCalibrationUpdated;
+            }
+
+            for (int i = 0; i < nextArUcoImageId; i++)
+            {
+                var dslrTexture = CalibrationDataHelper.LoadDSLRArUcoImage(i);
+                var headsetData = CalibrationDataHelper.LoadHeadsetData(i);
+
+                if (dslrTexture == null ||
+                    headsetData == null ||
+                    !ProcessArUcoData(headsetData, dslrTexture))
+                {
+                    Debug.LogWarning($"Failed to process dataset: {i}");
+                }
+                else
+                {
+                    CalibrationDataHelper.SaveDSLRArUcoDetectedImage(dslrTexture, i);
+                    CreateVisual(headsetData, i);
+                }
+            }
+        }
+
+        private void OnHeadsetCalibrationUpdated(byte[] data)
+        {
+            if(HeadsetCalibrationData.TryDeserialize(data, out var headsetCalibrationData))
+            {
+                this.headsetData = headsetCalibrationData;
+            }
+        }
+
+        private void Update()
+        {
+            if (feedImage != null &&
+                feedImage.texture == null)
+            {
+                feedImage.texture = CompositorWrapper.GetDSLRFeed();
+            }
+
+            if (Input.GetKeyDown(KeyCode.Space))
```

Comment:
Not a fan of hardcoding inputs like this.

The MRTK uses a platform input data provider abstraction. It might be best to run this stuff on an event response.

Further, this code will run continuously if you hold the space bar. That might not be ideal.

Reply:
If i'm interpreting Unity's documentation correctly, GetKeyDown will only get called for the first frame the button is pressed: https://docs.unity3d.com/ScriptReference/Input.GetKeyDown.html

Right now this code is intended to run in the editor and isn't meant for an end user application. We also don't want this code to take any dependencies on the foundational/core MRTK components, which may be required for the MRTK input abstraction.

I think for now this change may be beyond the scope of this review. This initial checkin will still need some reworking once qr code detection dlls are available to the public.